libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -310,123 +143,61 @@ - (void)enumerateDelegates:(void (^)(id<SFUserAccountManagerDelegate>))block
-}
-
-- (BOOL)autocreateAnonymousUser {
-    return [[[NSBundle mainBundle] objectForInfoDictionaryKey:kSFUserAccountAutocreateAnonymousUser] boolValue];
-}
-
 - (BOOL)isCurrentUserAnonymous {
-    return [[self class] isUserAnonymous:self.currentUser];
+    return self.currentUser == nil;
 }
 
-- (SFUserAccount *)anonymousUser {
-    if (nil == _anonymousUser) {
-        for (SFUserAccountIdentity *identity in self.allUserIdentities) {
-            if ([identity isEqual:self.anonymousUserIdentity]) {
-                _anonymousUser = [self userAccountForUserIdentity:identity];
-                break;
-            }
-        }
+-(NSMutableDictionary *)userAccountMap {
+    if(!_userAccountMap || _userAccountMap.count < 1) {
+        [self reload];
     }
-    return _anonymousUser;
+    return _userAccountMap;
 }
 
-- (void)setupAnonymousUser:(BOOL)supportsAnonymousUser autocreateAnonymousUser:(BOOL)autocreateAnonymousUser {
-
-    // If there is no current user but the application support anonymous user
-    // and wants it to be created automatically, then create it now.
-    if (supportsAnonymousUser && autocreateAnonymousUser && nil == self.anonymousUser) {
-        [self log:SFLogLevelInfo msg:@"Creating anonymous user"];
-        [self enableAnonymousAccount];
-        if (nil == self.currentUser) {
-            self.currentUser = self.anonymousUser;
-        }
-    }
+- (void)setUserAccountMap:(NSMutableDictionary *) userAccountMap {
+    _userAccountMap = userAccountMap;
 }
 
-- (void)enableAnonymousAccount {
-    if (nil == self.anonymousUser) {
-        self.anonymousUser = [[SFUserAccount alloc] initWithIdentifier:[self uniqueUserAccountIdentifier] clientId:self.oauthClientId];
-        SFOAuthCredentials *creds = self.anonymousUser.credentials;
-        creds.domain = self.loginHost;
-        creds.redirectUri = self.oauthCompletionUrl;
-        creds.clientId = self.oauthClientId;
-        creds.identityUrl = [NSURL URLWithString:[NSString stringWithFormat:@"https://does.not.matter.domain-does-not-exist/id/%@/%@", self.anonymousUserIdentity.orgId, self.anonymousUserIdentity.userId]];
-        creds.communityUrl = [NSURL URLWithString:@"https://who-cares.domain-does-not-exist"];
-        creds.instanceUrl = [NSURL URLWithString:@"https://who-cares.domain-does-not-exist"];
-        creds.accessToken = [NSUUID UUID].UUIDString;
-        [self addAccount:self.anonymousUser];
-        [self saveAccounts:nil];
-    }
+-(id<SFUserAccountPersister>)accountPersister {
+    return _accountPersister;
 }
 
-- (void)disableAnonymousAccount {
-    NSError *error;
-    if (![self deleteAccountForUser:self.anonymousUser error:&error]) {
-        [self log:SFLogLevelError format:@"Unable to delete the anonymous user: %@", [error localizedDescription]];
-    }
-    self.anonymousUser = nil;
+- (void)setAccountPersister:(id<SFUserAccountPersister>) persister {
+    [_accountsLock lock];
```

Comment:
Does this really need to be locked? Do we anticipate this being called in numerous places?